AirCasting/SessionViews/ChartView.swift

```diff
@@ -70,19 +58,79 @@ class UI_PollutionChart: UIView {
 
 struct ChartView: UIViewRepresentable {
     
+    @ObservedObject var stream: MeasurementStreamEntity
+    var thresholds: SensorThreshold
+    
     typealias UIViewType = UI_PollutionChart
     
     func makeUIView(context: Context) -> UI_PollutionChart {
         UI_PollutionChart()
     }
     
     func updateUIView(_ uiView: UI_PollutionChart, context: Context) {
+        let chartCreator = ChartEntriesCreator(stream: stream)
+        var entries = chartCreator.generateEntries()
+        
+        if entries.isEmpty {
+            return()
+        } else {
+            entries.sort { (e1, e2) -> Bool in
+                e1.x < e2.x
+            }
+        }
+        
+        let dataSet = LineChartDataSet(entries: entries)
+        let data = LineChartData(dataSet: dataSet)
+        uiView.lineChartView.data = data
+        
+        //format data labels
+        formatData(data: data)
+        
+        //format line and dots
+        formatDataSet(dataSet: dataSet)
+    }
+    
+    private func formatData(data: LineChartData) {
+        let formatter = NumberFormatter()
+        formatter.maximumFractionDigits = 0
+        data.setValueFormatter(DefaultValueFormatter(formatter: formatter))
+        data.setValueFont(UIFont.muli(size: 12))
+        data.setValueTextColor(UIColor.aircastingGray)
+    }
+    
+    private func formatDataSet(dataSet: LineChartDataSet) {
+        //dots colors
+        dataSet.circleColors = generateColorsSet(for: dataSet.entries)
+        dataSet.drawCircleHoleEnabled = false
+        dataSet.circleRadius = 6
+        
+        //line color
+        dataSet.setColor(UIColor.aircastingGray.withAlphaComponent(0.7))
+    }
+    
+    private func generateColorsSet(for entries: [ChartDataEntry]) -> [UIColor] {
+        var colors: [UIColor] = []
+        for entry in entries {
+            switch Int32(entry.y) {
```

Comment:
there seams to be a common problem to have this values of different types and we seams to change this very freely but the int conversion can lead to strange bugs in strange places. 
On a 32-bit platform, Int is the same size as Int32.
On a 64-bit platform, Int is the same size as Int64.